### ISSUE TITLE
Transcode or direct stream

### ISSUE THREAD

**Issue by microdockman** [stashapp/stash#2557]
This shouldn't be a bug, but it's a bit troubling

all my media files are h264 ，however, some videos are forced to be encoded during playback,
I found that frame rate determines whether or not to encode  （60fps encoded && 30fps direct)
Based on my poor host performance, I can't afford to encode playback

Whether the direct Stream can be forcibly enabled

**Comment by WithoutPants** [comment]
Direct stream only works where the browser supports it. If the browser cannot directly stream the file, then it falls back to encoding. You can verify this by trying to stream the file directly using the `Stream` link in file info in your browser. The stream link looks like `http://localhost:9999/scene/<sceneid>/stream`. If you can successfully view a file this way in your browser but not in the stash scene player then it is indeed a bug. Usually though, the browser will either try to download the file or will only be able to stream the audio without the video or vice versa.

**Comment by microdockman** [comment]
> Direct stream only works where the browser supports it. If the browser cannot directly stream the file, then it falls back to encoding. You can verify this by trying to stream the file directly using the `Stream` link in file info in your browser. The stream link looks like `http://localhost:9999/scene/<sceneid>/stream`. If you can successfully view a file this way in your browser but not in the stash scene player then it is indeed a bug. Usually though, the browser will either try to download the file or will only be able to stream the audio without the video or vice versa.


There is a file that validates this problem, I don't know if it will enable transcoding for audio, but compared to other files, this might tell something

1920 x 1080
59.94 fps
51.94 Mbps/s
h264
pcm_s16be

the transcoding will be started in Stash, 
but http:// localhost: 9999/SCENE/<Sceneid>/Stream will be played directly.

**Comment by microdockman** [comment]
> > Direct stream only works where the browser supports it. If the browser cannot directly stream the file, then it falls back to encoding. You can verify this by trying to stream the file directly using the `Stream` link in file info in your browser. The stream link looks like `http://localhost:9999/scene/<sceneid>/stream`. If you can successfully view a file this way in your browser but not in the stash scene player then it is indeed a bug. Usually though, the browser will either try to download the file or will only be able to stream the audio without the video or vice versa.
> 
> There is a file that validates this problem, I don't know if it will enable transcoding for audio, but compared to other files, this might tell something
> 
> 1920 x 1080 59.94 fps 51.94 Mbps/s h264 pcm_s16be
> 
> the transcoding will be started in Stash, but http:// localhost: 9999/SCENE//Stream will be played directly.

stashapp/stash:development
google chrome 100.0.4896.127
windows10

**Comment by WithoutPants** [comment]
> There is a file that validates this problem, I don't know if it will enable transcoding for audio, but compared to other files, this might tell something
> 
> 1920 x 1080 59.94 fps 51.94 Mbps/s h264 pcm_s16be
> 
> the transcoding will be started in Stash, but http:// localhost: 9999/SCENE//Stream will be played directly.

Does playing directly via `/stream` include the audio?

**Comment by microdockman** [comment]
> > There is a file that validates this problem, I don't know if it will enable transcoding for audio, but compared to other files, this might tell something
> > 1920 x 1080 59.94 fps 51.94 Mbps/s h264 pcm_s16be
> > the transcoding will be started in Stash, but http:// localhost: 9999/SCENE//Stream will be played directly.
> 
> Does playing directly via `/stream` include the audio?

Yes, included audio plays within the web page, just like potplayer plays

**Comment by WithoutPants** [comment]
Reproduced. Thank you for the report.

**Comment by WithoutPants** [comment]
Unfortunately, there's no easy fix for this issue. My attempts to add PCM as a valid codec failed because some browsers do not support PCM, but there is no way to check in the browser that the audio codec is invalid, so that the video plays without playing the audio. I've put this in the backlog for now to solve at a later date. It will require a larger change.

**Comment by microdockman** [comment]
> Unfortunately, there's no easy fix for this issue. My attempts to add PCM as a valid codec failed because some browsers do not support PCM, but there is no way to check in the browser that the audio codec is invalid, so that the video plays without playing the audio. I've put this in the backlog for now to solve at a later date. It will require a larger change.

many thanks..

**Comment by microdockman** [comment]
Determine the problem: 
The problem is caused by the audio format. up to now, only aac audio format supports direct streaming.  
other audio formats such as ac3 will force transcoding
It has nothing to do with the video bit rate

**Comment by feederbox826** [comment]
There's no real way to accomplish this without deconstructing the file into audio/ video streams and listening for support manually. Transcode is unfortunately the only way

**Comment by HarletKing** [comment]
@microdockman While not a solution, I've had the same problem with PCM encoded videos. I resorted to setting up Tdarr to monitor all added files for the codec and have them transcoded to Opus (more efficient than AAC, well supported on clients). Since it's only audio, you don't need a GPU for an efficient transcode.
